src/CodeGenerators/SettingsGen/Models/Attributes/ParameterAttribute.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.CodeGenerators.SettingsGen.Models.Attributes
+{
+	/// <summary>
+	/// Attribute representing a parameter in the Settings.xml file
+	/// </summary>
+	[AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+	public sealed class ParameterAttribute : Attribute
+	{
+		/// <summary>
+		/// Name of the parameter
+		/// </summary>
+		public string Name { get; set; } = string.Empty;
+
+		/// <summary>
+		/// Value of the parameter
+		/// </summary>
+		public string Value { get; set; } = string.Empty;
+
+		/// <summary>
+		/// MustOverride field of the Parameter element
+		/// </summary>
+		public string? MustOverride { get; set; }
+
+		/// <summary>
+		/// IsEncrypted field of the Parameter element
+		/// </summary>
+		public string? IsEncrypted { get; set; }
+
+		/// <summary>
+		/// Type field of the Parameter element
+		/// </summary>
+		public string? Type { get; set; }
```

Comment:
Why this field is needed?

Reply:
This is a valid field in service fabric settings xml

Reply:
But why it's needed on an attribute? Could not we get the type from property type?

Reply:
Type field setting and property type are not the same thing, it can only be a number of predefined values, but didn't think it was worth creating an enum for it given its low usage

Reply:
And what's is the use case for this field? when would we need to override it?

Reply:
Basically you would state if its plaintext or encrypted, we haven't had to use it but keeping it in there to allow others to use it

Reply:
https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-service-model-schema-elements#xml-source-158